addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,442 @@
+                    res[invoice] = {
+                        "success": False,
+                        "error": error_message,
+                        "blocking_level": ("404" in error_codes) and "warning" or "error",
+                    }
+            if not response.get("error"):
+                json_dump = json.dumps(response.get("data"))
+                json_name = "%s_cancel_einvoice.json" % (invoice.name.replace("/", "_"))
+                attachment = self.env["ir.attachment"].create({
+                    "name": json_name,
+                    "raw": json_dump.encode(),
+                    "res_model": "account.move",
+                    "res_id": invoice.id,
+                    "mimetype": "application/json",
+                })
+                res[invoice] = {"success": True, "attachment": attachment}
+        return res
+
+    def _l10n_in_validate_partner(self, partner, is_company=False):
+        self.ensure_one()
+        message = []
+        if is_company and partner.country_id.code != "IN":
+            message.append(_("\n- Country should be India"))
+        if not re.match("^.{3,100}$", partner.street or ""):
+            message.append(_("\n- Street required min 3 and max 100 characters"))
+        if partner.street2 and not re.match("^.{3,100}$", partner.street2):
+            message.append(_("\n- Street2 should be min 3 and max 100 characters"))
+        if not re.match("^.{3,100}$", partner.city or ""):
+            message.append(_("\n- City required min 3 and max 100 characters"))
+        if not re.match("^.{3,50}$", partner.state_id.name or ""):
+            message.append(_("\n- State required min 3 and max 50 characters"))
+        if partner.country_id.code == "IN" and not re.match("^[0-9]{6,}$", partner.zip or ""):
+            message.append(_("\n- Zip code required 6 digits"))
+        if partner.phone and not re.match("^[0-9]{10,12}$", self.env["account.move"]._extract_digits(partner.phone)):
+            message.append(_("\n- Mobile number should be minimum 10 or maximum 12 digits"))
+        if partner.email and (
+            not re.match(r"^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$", partner.email)
+            or not re.match("^.{6,100}$", partner.email)
+        ):
+            message.append(_("\n- Email address should be valid and not more then 100 characters"))
+        return message
+
+    def _get_l10n_in_edi_saler_buyer_party(self, move):
+        return {
+            "seller_details": move.company_id.partner_id,
+            "dispatch_details": move._l10n_in_get_warehouse_address() or move.company_id.partner_id,
+            "buyer_details": move.partner_id,
+            "ship_to_details": move.partner_shipping_id or move.partner_id,
+        }
+
+    @api.model
+    def _get_l10n_in_edi_partner_details(self, move, partner,
+        vat=True, phone_and_email=True, pos=False, check_treatment=False):
```

Comment:
This method looks quite messy. Also, there is nothing on its name indicating it's about invoices meaning if you plan to use it one day for something else, this will be unclear. This method should be either partner-specific, either-invoice specific but not mixed like that

Reply:
okay, trying convert into partner-specific